tools/clang/tools/dxcompiler/dxclinker.cpp

```diff
@@ -140,6 +211,24 @@ DxcLinker::RegisterLibrary(_In_opt_ LPCWSTR pLibName, // Name of the library.
         pBlob->GetBufferPointer(), pBlob->GetBufferSize(), pModule,
         pDebugModule, m_Ctx, m_Ctx, DiagStream));
 
+
+    // add an entry into the library to compiler version part map
+    const hlsl::DxilContainerHeader *pHeader = hlsl::IsDxilContainerLike(
+        pBlob->GetBufferPointer(), pBlob->GetBufferSize());
+    const DxilPartHeader *pDPH = hlsl::GetDxilPartByType(
+        pHeader, hlsl::DxilFourCC::DFCC_CompilerVersion);
+    if (pDPH) {
+        const hlsl::DxilCompilerVersion *pDCV =
+            (const hlsl::DxilCompilerVersion *)(pDPH + 1);
+        // If the compiler version string is non-empty, add the struct to the
+        // map
+        bool success =
+            AddCompilerVersionMapEntry(pLibName, pDCV, pDPH->PartSize);
+        if (!success) {
+        return E_INVALIDARG;
```

Comment:
minor indent issue. 

Additionally, the `success` variable is only used for this check and might as well be `if(!AddCompilerVersionMapEntry...`